### DISCUSSION TITLE
zsh process remains alive after closing window

### DISCUSSION THREAD

**Discussion by JoseMenta** [ghostty-org/ghostty#3790] [Q&A] [answered]
Hello, I've just started using Ghostty on macOS and I've really liked it so far. 

I've noticed that when closing the window with the top red button, the _zsh_ process associated with that window is not killed. As a result, when I open another window without quitting the app, I can see all the processes of the previous windows with their new TTY for each one. 
For example, when I first open Ghostty, running `ps` shows the following output:
```
> ps -o pid,tty,time,command,stat
  PID TTY           TIME COMMAND   STAT
  856 ttys000    0:00.08 -/bin/zsh S
```
Then, I close the window with the red button and open Ghostty again, and the result is the following:
```
> ps -o pid,tty,time,command,stat
  PID TTY           TIME COMMAND   STAT
  856 ttys000    0:00.09 -/bin/zsh S+
 1197 ttys001    0:00.04 -/bin/zsh S
```
In this case, the process I expected to be terminated is the one with pid 856. This also happens when using `Shift + Command + w` (the Close Window shortcut) but it does not happen when using `Command + w` (the Close shortcut), where the process of the closed window is killed. The remaining processes are only killed when quitting the Ghostty app. 

Is there any way to change the configuration so that closing the window with the red button also kills the _zsh_ process associated with that window?

Chip: Intel
OS: macOS 15.2
Install method: download binary (.dmg)

**Comment by mitchellh** [comment, marked as answer]
I haven't heard any other reports of this, and we've revamped a lot of our process management significantly since report. Please try later versions (particularly tip) and open a new discussion if this still occurs.
